LearningJourney/Modules/Features/JLibrary/JLibraryTests/Sources/Data/LibraryParserTests.swift

```diff
@@ -0,0 +1,42 @@
+import XCTest
+@testable import JLibrary
+
+final class LibraryParserTests: XCTestCase {
+    // MARK: - System under test
+
+    private let sut = LibraryParser()
+
+    // MARK: - Tests
+
+    func testParseDecodableDataShouldSucceed() throws {
+        // Given
+        let decodable = "aDecodableString"
+        let encodedData = try XCTUnwrap(decodable.data(encoding: .utf8))
+        let expectedDecodedData: Result<String, Error> = .success(decodable)
+
+        // When
+        let actualParsedValue = sut.parse(encodedData)
+
+        // Then
+        
+        XCTAssertEqual(actualParsedValue, expectedDecodedData)
+    }
+
+    func testParseUndecodableDataShouldFail() throws {
+        // Given
+        let invalidJson = "{" // any invalid JSON is undecodable
+        let encodedData = try XCTUnwrap(invalidJson.data(using: .utf8))
+
+        // When
+        let actualResult = sut.parse(encodedData)
+
+        // Then
+        XCTAssertThrowsError(try actualResult.get(), "TODO Message!")
```

Comment:
Please resolve this TODO
```suggestion
        XCTAssertThrowsError(try actualResult.get(), "Expected decoding to fail, but got \(actualResult)")
```